packages/runtime/container-runtime/src/opLifecycle/remoteMessageProcessor.ts

```diff
@@ -116,10 +122,17 @@ export class RemoteMessageProcessor {
 
 		// Do a final unpack of runtime messages in case the message was not grouped, compressed, or chunked
 		unpackRuntimeMessage(message);
-		return {
-			messages: [message as InboundSequencedContainerRuntimeMessageOrSystemMessage],
-			batchStartCsn,
-		};
+		this.processorBatch.push(
+			message as InboundSequencedContainerRuntimeMessageOrSystemMessage,
+		);
+		if (this.batchStartCsn === undefined) {
```

Comment:
This needs to be more obvious why, in terms of the code or comments.  Two ideas:

1. Change `getAndUpdateBatchStartCsn` to return `{ batchStartCsn: number,  endOfBatch: boolean }`, and key off `endOfBatch` here. Probably rename that helper to something like `updateBatchState` too?
2. Just add a comment here explaining that this condition means we've processed all messages in a batch, so we're at the end of the batch and should return all the messages together.